django/contrib/gis/gdal/raster/source.py

```diff
@@ -151,6 +184,13 @@ def __init__(self, ds_input, write=False):
         else:
             raise GDALException('Invalid data source input type: "{}".'.format(type(ds_input)))
 
+    def __del__(self):
```

Comment:
A task in https://code.djangoproject.com/ticket/23919 is "Evaluate replacement of custom `__del__` methods by  [weakref.finalize()](https://docs.python.org/3/library/weakref.html#weakref.finalize)". I wonder if you have any input about that?